scalafix-diff/js/src/main/scala/scalafix/diff/DiffUtils.scala

```diff
@@ -29,12 +29,18 @@ object DiffUtils {
       "",
       "",
       js.Dynamic.literal("context" -> contextSize))
+
     def trimHeader(line: String) =
       if (line.startsWith("+++") || line.startsWith("---")) line.trim else line
-    diff
-      .split("\n")
-      .drop(1) // remove ==== separator
-      .map(trimHeader) // remove whitespaces at the end of headers
-      .mkString("\n")
+
+    def removeNewlineDiff(diffLines: Array[String]) =
+      diffLines.filterNot(_ == "\\ No newline at end of file")
+
+    removeNewlineDiff(
+      diff
+        .split("\n")
```

Comment:
s/.split(..)/.lines/ will return an iterator, which is lazy and avoids the creation of an intermediary array in all steps.